src/ruuvitag_ble/parser.py

```diff
@@ -23,14 +25,25 @@ def _start_update(self, service_info: BluetoothServiceInfo) -> None:
             return None
 
         data_format = raw_data[0]
-        if data_format != 0x05:
+        if data_format not in (0x03, 0x05):
             _LOGGER.debug("Data format not supported: %s", raw_data)
             return
 
-        decoder = DataFormat5Decoder(raw_data)
+        decoder_classes: dict[
+            int,
+            Type[Union[DataFormat3Decoder, DataFormat5Decoder]],
+        ] = {
+            0x03: DataFormat3Decoder,
+            0x05: DataFormat5Decoder,
+        }
+
+        decoder: Union[DataFormat3Decoder, DataFormat5Decoder] = decoder_classes.get(
+            data_format,
+            DataFormat5Decoder,
+        )(raw_data)
+
         # Compute short identifier from MAC address
-        # (preferring the MAC address the tag broadcasts).
-        identifier = short_address(decoder.mac or service_info.address)
+        identifier = short_address(service_info.address)
```

Comment:
This change shouldn't be made in this PR. You can add a `mac = None` class-level attribute to `DataFormat3Decoder` (since indeed, DF3 doesn't broadcast the MAC).